### ISSUE TITLE
Deploy to AWS Lambda for Node.JS / WebStorm

### ISSUE THREAD

**Issue by kiiadi** [aws/aws-toolkit-jetbrains#187]
The core plugin exposes two extension-points that enable deployment of an AWS Lambda.
* `aws.toolkit.lambda.packager` builds the appropriate zip-file for the specific runtime; extensions implement [`LambdaPackager`](software.aws.toolkits.jetbrains.services.lambda.LambdaPackager).
* `aws.toolkit.lambda.handlerResolver` contains logic for marking `PsiElements` as potential lambda handler methods (this is used to create a gutter icon which exposes the 'deploy to Lambda' action). Extensions implement [`LambdaHandlerResolver`](software.aws.toolkits.jetbrains.services.lambda.LambdaHandlerResolver)

We need an implementations of these for Node.js that follow the AWS Lambda packaging instructions : https://docs.aws.amazon.com/lambda/latest/dg/nodejs-create-deployment-pkg.html

**Comment by kiiadi** [comment]
The beginnings of this is in this branch: https://github.com/aws/aws-toolkit-jetbrains/tree/feature/nodejs-support

**Comment by fcasals** [comment]
please add Webstorm to the roadmap of this plugin!
🙏

**Comment by kiiadi** [comment]
Would love to get some feedback on the progress we've made towards WebStorm support. An early preview build of the WebStorm functionality can be found here : 

[aws-jetbrains-toolkit-1.6-webstorm-beta-3.zip](https://github.com/aws/aws-toolkit-jetbrains/files/3562136/aws-jetbrains-toolkit-1.6-webstorm-beta-3.zip)

Download the zip and use the "Install from Disk..." option to install the plugin into WebStorm or your Node enabled IntelliJ IDE.

Alternatively if you'd prefer to build from source yourself do the following:

```
git clone https://github.com/aws/aws-toolkit-jetbrains.git
cd aws-toolkit-jetbrains
git checkout feature/webstorm-support
./gradlew buildPlugin
```

This will produce a zip in the `build/distribution` folder - use this to install from disk.

Currently this supports the same feature set as Python & Java, namely:
* Create new project
* Step-through debug locally
* Deploy serverless application

Similar to the other language-support in the toolkit there are some required dependencies see the documentation [here](https://docs.aws.amazon.com/toolkit-for-jetbrains/latest/userguide/setup-toolkit.html).

Please post back to this issue if you have feedback.

**Comment by charliecode** [comment]
@kiiadi Downloaded aws-jetbrains-toolkit-1.6-webstorm-beta-3.zip and am having trouble installing. I'll give more details about issues but first let me make sure I'm trying the correct file. Which file should be selected within zip folder to install plugin?

**Comment by abrooksv** [comment]
@charliecode In the IDE, when you select Install from Disk, select the zip file itself.

**Comment by charliecode** [comment]
Major bug: Webstorm 2019.2.2

Created a small Lambda Function within my already created app (not Serverless). When I choose "Create new AWS Lambda" and fill out all the appropriate fields, most notably "Handler:" and add where the handler is located, which is a single small file and click "Create Function" it creates a new .aws-sam folder at the root of my entire app. It contains not the single small lambda function I created but rather my entire app, causing my system to prompt me to "add" my entire app (all files) to git. The function is also never loaded to lambda because it's way too large. The file is successfully loaded to S3, but again, it's not the single Lambda function I created but rather my entire app.
